src/coreclr/nativeaot/Runtime/gcrhenv.cpp

```diff
@@ -1181,8 +1181,13 @@ bool GCToEEInterface::EagerFinalized(Object* obj)
     // Managed code should not be running.
     ASSERT(GCHeapUtilities::GetGCHeap()->IsGCInProgressHelper());
 
+    // the lowermost 2 bits are reserved for storing additional info about the handle
+    // we can use these bits because handle is at least 32bit aligned
+    // we also reserve the sign bit
```

Comment:
It is kind of odd to use the term "sign bit" for an unsigned integer type. Instead how about we say "highest order bit"?

Reply:
@AaronRobinsonMSFT - it looks like we will be changing the managed part of the implementation quite significantly. 

The comments are appropriate, but I think most of this code will need a rewrite.

This approach does not work well on x86. We keep finding issues with that. For example using the sign bit for tagging is not safe. The tests are passing (which means the native part that deals with COM is working correctly), but we are just lucky as the sign/highermost bit could be used by the pointer on x86, where pointer values can cover most of the 4Gb range, and then tagging would result in pointer corruption.  
Also it looks like with the new approach we do not need to involve `ConditionalWeakTable`, which seems nice.

I'll try sketching the new approach (likely this Friday).